### PR TITLE
Adding linter

### PR DESCRIPTION
**Pull request by Hartmannsson-lev**

### CODE REVIEW

**Comment by mehar-chauhan** [comment]
Jenkins pipeline: https://lev-ho-hmpo-ew2tlg1nl1-i-team-lev-i-docker-tlg1.np.ebsa.homeoffice.gov.uk/job/apps/job/lev-web-ui/job/LEV-458-eslint/6/console

File: eslint.config.mjs [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import globals from "globals";
+import pluginJs from "@eslint/js";
+import tseslint from "typescript-eslint";
+
+export default [
+  {
+    files: ["**/*.js"],
+    languageOptions: {
+      sourceType: "script",
+      globals: globals.browser
```
Comment (ashley-bickerstaff-homeoffice) [inline eslint.config.mjs:10]: I've run the linter and it was surfacing over 2000 problems... However, most of them are not actual problems. I know the scope of this story isn't to resolve the issues, but I think that the nature of the problems being identified (lots of "require is not defined" type errors) point to the configuration being incorrect.

The following suggestion resolves the vast majority of these. Combined with my other comment in the "ignores" section, it reduces the count to ~700.

```suggestion
      globals: {
        ...globals.browser,
        ...globals.node,
        ...globals.jest
      }
```

File: eslint.config.mjs [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    }
+  },
+  {
+    ignores: ['dist/']
```
Comment (ashley-bickerstaff-homeoffice) [inline eslint.config.mjs:14]: We need to exclude `/public` too as this contains generated files which we don't have lots of control over:

```suggestion
    ignores: ['dist/', 'public/']
```

**Review by ashley-bickerstaff-homeoffice — CHANGES_REQUESTED** [review]
Need to resolve the issues I've identified before it can be merged.

File: Makefile [hunk 3]
```diff
@@ -9,6 +9,9 @@ clean:
 install:
 	npm install
 
+lint:
+	npm install && npx eslint . --max-warnings=0 || exit 0
```
Comment (ashley-bickerstaff-homeoffice) [inline Makefile:13]: Can we change this to use the npm runner? I think the following will work.

```suggestion
	npm install && npm run lint --max-warnings=0 || exit 0
```

**Pull request merged**